src/SourceBuild/content/eng/build.targets

```diff
@@ -0,0 +1,54 @@
+<Project>
+  <UsingTask AssemblyFile="$(SdkArchiveDiffTasksAssembly)" TaskName="GetValidArchiveItems" />
+  <UsingTask AssemblyFile="$(SdkArchiveDiffTasksAssembly)" TaskName="GetClosestOfficialSdk" />
+  <UsingTask AssemblyFile="$(SdkArchiveDiffTasksAssembly)" TaskName="FindArchiveDiffs" />
+
+  <Target Name="ReportSdkArchiveDiffs"
+          AfterTargets="Build"
+          DependsOnTargets="DetermineSourceBuiltSdkVersion"
+          Condition="'$(ShortStack)' != 'true' AND '$(PortableBuild)' == 'true'" >
+          <!-- ShortStack doesn't produce an SDK, and GetClosestOfficialSdk doesn't support finding non-portable SDKs -->
+
+      <Message Text="Comparing built SDK against closest official build"
+               Importance="High"/>
+
+      <GetValidArchiveItems ArchiveItems="@(SdkTarballItem)"
+                            ArchiveName="dotnet-sdk">
+        <Output TaskParameter="ValidArchiveItems"
+                ItemName="_BuiltSdkArchivePath"/>
+      </GetValidArchiveItems>
+
+      <!-- There should only be 1 SDK archive -->
+      <Error Text="Multiple valid dotnet-sdk archives found."
+             Condition="'@(_BuiltSdkArchivePath->Count())' != '1'" />
+
+      <GetClosestOfficialSdk BuiltArchivePath="@(_BuiltSdkArchivePath)"
```

Comment:
I would be surprised if this fails to find a matching SDK...does this happen?

Reply:
Are you referring to the `Condition="'@(_BuiltSdkArchivePath)' != ''"` or something else? I think we error in the tasks before we'd hit the conditions for `GetClosestOfficialSdk` and `FindArchiveDiffs`, so I'll remove the conditions.

Reply:
Ahh, here: https://github.com/dotnet/installer/pull/18748/files#diff-a4e89432a27e7959cd783637704d06e20a30c549d96948cc59ccea3bb16c7796R33

I would be surprised if the returned `_ClosestOfficialSdkPath` is empty.

Reply:
Gotcha, yeah that should never happen, we'll error out in the task first. I removed those conditions.